src/components/Article.jsx

```diff
@@ -0,0 +1,48 @@
+import {useEffect, useState } from 'react';
+import {useParams} from 'react-router-dom'
+import {getArticleById} from '../api'
+
+const Article = () => {
+
+    let {article_id} = useParams();
+
+    const [article, setArticle] = useState(null);
+    const [isLoading, setLoading] = useState(true);
+
+    useEffect(() => {
+        setLoading(true);
+        getArticleById(article_id).then(({article}) => {
+            setArticle(article);
+            setLoading(false);
+        })
+    }, [])
+    if (!article) {
+        return <p className="error">No Articles Found</p>
+      }
```

Comment:
Probably wouldnt do the error handling like this - you'd do it in the catch in the useEffect
if you make a request, and the api says that id isnt there, it will reject, then you can handle that error.